core/platform/image/gae_image_services.py

```diff
@@ -79,9 +78,12 @@ def compress_image(image_content, scaling_factor):
                     MAX_RESIZE_DIMENSION_PX, float(max(width, height))))
             new_width = int(width * new_scaling_factor)
             new_height = int(height * new_scaling_factor)
-        return images.resize(
-            image_data=image_content,
-            width=min(new_width, MAX_RESIZE_DIMENSION_PX),
-            height=min(new_height, MAX_RESIZE_DIMENSION_PX))
+        img = img.resize((min(new_width, MAX_RESIZE_DIMENSION_PX),
+                          min(new_height, MAX_RESIZE_DIMENSION_PX)))
+        with io.BytesIO() as output:
+            img.save(output, format='PNG')
```

Comment:
Is it guaranteed that the original image is of PNG format? If not, your changes are changing the behaviour of the original function, which they should not do.

One other thing I noticed is that this function seems to have no tests. I suggest doing the following, given that you want to change it carefully while preserving the behaviour:

(a) revert your changes
(b) write comprehensive tests to cover all the behaviour of this function -- try a range of possible inputs and outputs. Note that you can store image files for tests in core/tests/data.
(c) make your changes and ensure the tests pass. Verify that your tests would have caught the issue mentioned above.

Reply:
Done